Content.Server/GameObjects/Components/Explosion/ExplosiveComponent.cs

```diff
@@ -30,14 +30,16 @@ public override void ExposeData(ObjectSerializer serializer)
 
         public bool Explosion()
         {
-            //Prevent adjacent explosives from infinitely blowing each other up.
-            if (_beingExploded) return true;
-            _beingExploded = true;
-
-            Owner.SpawnExplosion(DevastationRange, HeavyImpactRange, LightImpactRange, FlashRange);
-
-            Owner.Delete();
-            return true;
+            if (Exploding)
+            {
+                return false;
+            }
+            else
+            {
+                Exploding = true;
+                Owner.SpawnExplosion(DevastationRange, HeavyImpactRange, LightImpactRange, FlashRange);
+                return true;
```

Comment:
I think that should be handled by the ActSystem with handleExplosion.

Reply:
The previous behavior of the component is that the entity is deleted when it explodes. If you leave it up to IExAct (called from ActSystem), then isn't it only going to be destroyed as a side-effect of the explosion being applied to the entity itself? So if for some reason the entity doesn't take damage, it wouldn't be deleted?

Better IMO to ensure it's deleted regardless of what happens in ActSystem. Unless you can elaborate on why you think it's better to leave it up to IExAct. If there REALLY is a use case for having the entity not always be destroyed, we can add a flag to ExplosiveComponent to not delete the entity on explosion (defaulting to deleting it).

Reply:
I can see some cases where it works
Let's say we have a meteor entity that after touching the station and after some ticks it explodes, instead of disappearing, it remains there, and maybe someone can extract something useful from it.
Or in the case of a fuel tank some ss13 servers change the sprite to a destroyed fuel tank. And yes this can be implemented  the way a wall spawns a girder after receiving enough damage. But that would be another prototype entry. Maybe a gas canister explodes but the broken tank still remains.
If the explosion doesn't decide if the entity is deleted, you can still 'play' with it, just in another state.

I think .Delete() is more like a bandaid. It is better to trust the system in charge of handling those types of events. The flag option sounds good to me.

Reply:
ok, that makes sense, deleting in ALL cases would be an issue. Let's go with a flag.

Reply:
For now I will keep it as the original , until I/someone gets to it